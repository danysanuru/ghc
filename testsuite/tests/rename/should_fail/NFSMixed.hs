{-# LANGUAGE NoFieldSelectors #-}
module NFSMixed where
data Foo = Foo { foo :: Int, bar :: String }
data Bar = Bar { foo :: Int, bar' :: String }

foo :: Int
foo = 0

test = \x -> x { foo = 0 }
