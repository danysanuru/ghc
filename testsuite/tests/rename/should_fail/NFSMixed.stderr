
NFSMixed.hs:9:18:
    Ambiguous occurrence ‘foo’
    It could refer to
       either the field ‘foo’, defined at NFSMixed.hs:4:18
           or the field ‘foo’, defined at NFSMixed.hs:3:18
           or ‘NFSMixed.foo’, defined at NFSMixed.hs:7:1